operator/informer_kubernetes.go

```diff
@@ -22,24 +22,38 @@ var (
 // KubernetesBasedInformer is a k8s apimachinery-based informer. It wraps a k8s cache.SharedIndexInformer,
 // and works most optimally with a client that has a Watch response that implements KubernetesCompatibleWatch.
 type KubernetesBasedInformer struct {
-	ErrorHandler        func(context.Context, error)
 	SharedIndexInformer cache.SharedIndexInformer
 	// HealthCheckIgnoreSync will set the KubernetesBasedInformer HealthCheck to return ok once the informer is started,
 	// rather than waiting for the informer to finish with its initial list sync.
 	// You may want to set this to `true` if you have a particularly long initial sync period and don't want readiness checks failing.
 	HealthCheckIgnoreSync bool
 	schema                resource.Kind
 	runContext            context.Context
+	eventTimeout          time.Duration
+	errorHandlerFn        func(context.Context, error)
 	healthCheckName       string
 }
 
-type KubernetesBasedInformerOptions struct {
+// InformerOptions are generic options for all Informer implementations.
+type InformerOptions struct {
 	// ListWatchOptions are the options for filtering the watch based on namespace and other compatible filters.
 	ListWatchOptions ListWatchOptions
 	// CacheResyncInterval is the interval at which the informer will emit CacheResync events for all resources in the cache.
 	// This is distinct from a full resync, as no information is fetched from the API server.
-	// An empty value will disable cache resyncs.
```

Comment:
I think we should still allow `CacheResyncInterval` being zero to disable cache resync. It looks like we still do, but this line of documentation is replaced.

Reply:
Fixed!